src/lib/core/Renderer.ts

```diff
@@ -332,19 +350,24 @@ class Renderer {
    * @param id - The ID of the capability to enable
    */
   enable(id: GLenum): void {
-    if (this.state[id] === true) return
+    if (this.state[id as unknown as keyof WTCGLRendererState] === true) return
+    // these typings are ugly I just couldn't find a way to map the rendering context to the state without major refactor of the  interface and its uses
+    if (this.state[id as unknown as keyof WTCGLRendererState] === true) return
+
     this.gl.enable(id)
-    this.state[id] = true
+    this.state[id as unknown as keyof WTCGLRendererState] = true as never
```

Comment:
This was the hardest part of this class. The state of things is controlled internally so the types are specific but at the same time we still `enable/disable` the native properties so there has to be some kind of conversion which makes this super nasty. `ts-ignore` is problematic because it confuses the type checker, it doesn't know the source of the type anymore. It might be okay here though, this feels pretty self contained.